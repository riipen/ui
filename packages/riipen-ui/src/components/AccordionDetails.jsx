import clsx from "clsx";
import PropTypes from "prop-types";
import React from "react";

const AccordionDetails = props => {
  const { classes, ...other } = props;

  return (
    <React.Fragment>
      <div className={clsx("root", classes)} {...other} />
      <style jsx>{`
        .root {
          display: flex;
        }
      `}</style>
    </React.Fragment>
  );
};

AccordionDetails.propTypes = {
  /**
   * The content of the accordion details.
   */
  children: PropTypes.node,

  /**
   * List of additional classes to apply to this component.
   */
  classes: PropTypes.array
};

AccordionDetails.displayName = "AccordionDetails";

export default AccordionDetails;
