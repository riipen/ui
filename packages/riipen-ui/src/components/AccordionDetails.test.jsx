import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import AccordionDetails from "./AccordionDetails";

describe("AccordianDetails", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders without errors", () => {
    let error;
    try {
      mount(<AccordionDetails />);
    } catch (e) {
      error = e;
    }

    expect(error).toEqual(undefined);
  });

  it("snapshot test", () => {
    const wrapper = mount(<AccordionDetails />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe("children prop", () => {
    it("displays children", () => {
      const child = <h5>Accordian Details</h5>;
      const wrapper = mount(<AccordionDetails children={child} />);

      expect(wrapper.contains(child)).toEqual(true);
    });
  });

  describe("classes prop", () => {
    it("applies classes to the root node of AccordianDetails", () => {
      const classes = ["classOne", "classTwo"];

      const wrapper = mount(<AccordionDetails classes={classes} />);

      expect(
        wrapper
          .find("AccordionDetails")
          .childAt(0)
          .hasClass(classes[0])
      ).toEqual(true);
      expect(
        wrapper
          .find("AccordionDetails")
          .childAt(0)
          .hasClass(classes[1])
      ).toEqual(true);
    });

    it("gives an error when classes are provided as one string", () => {
      const classes = "classOne classTwo";
      const errors = jest.spyOn(console, "error").mockImplementation();

      mount(<AccordionDetails classes={classes} />);

      expect(errors).toHaveBeenCalled();
    });
  });
});
