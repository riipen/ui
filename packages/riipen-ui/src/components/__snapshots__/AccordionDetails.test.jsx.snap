// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordianDetails renders correct snapshot 1`] = `
<AccordionDetails>
  <div
    className="jsx-1644707333 root"
  />
  <JSXStyle
    id="1644707333"
  />
</AccordionDetails>
`;
